File: endpoints/src/main/java/com/cromoteca/samples/endpoints/JacksonEndpoint.java
package com.cromoteca.samples.endpoints;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.vaadin.flow.server.auth.AnonymousAllowed;
import dev.hilla.Endpoint;

@Endpoint
@AnonymousAllowed
public class JacksonEndpoint {
  public void dummy(Entity entity, User user, Product product) {}

  @JsonIgnoreProperties({"ignoredByJsonIgnoreProperties"})
  public static class Entity {
    @JsonIgnore private String ignoredByJsonIgnore;

    @JsonProperty("renamedByJsonProperty")
    private String toBeRenamedByJsonProperty;

    private String ignoredByJsonIgnoreProperties;

    public String getIgnoredByJsonIgnore() {
      return ignoredByJsonIgnore;
    }

    public void setIgnoredByJsonIgnore(String ignoredByJsonIgnore) {
      this.ignoredByJsonIgnore = ignoredByJsonIgnore;
    }

    public String getToBeRenamedByJsonProperty() {
      return toBeRenamedByJsonProperty;
    }

    public void setToBeRenamedByJsonProperty(String toBeRenamedByJsonProperty) {
      this.toBeRenamedByJsonProperty = toBeRenamedByJsonProperty;
    }

    public String getIgnoredByJsonIgnoreProperties() {
      return ignoredByJsonIgnoreProperties;
    }

    public void setIgnoredByJsonIgnoreProperties(String ignoredByJsonIgnoreProperties) {
      this.ignoredByJsonIgnoreProperties = ignoredByJsonIgnoreProperties;
    }
  }

  @JsonIgnoreProperties(
      value = {"password"},
      allowSetters = true)
  public static class User {
    public String name;
    public String password;
  }

  @JsonIgnoreProperties(
      value = {"id"},
      allowGetters = true)
  public class Product {
    public String id;
    public String name;
  }
}

File: generator/src/test/resources/com/cromoteca/samples/endpoints/JacksonEndpoint.ts
import client from '../../../../connect-client.default.js';
import type Entity from './JacksonEndpoint/Entity.js';
import type Product from './JacksonEndpoint/Product.js';
import type User from './JacksonEndpoint/User.js';
import { type EndpointRequestInit } from '@vaadin/hilla-frontend';

async function dummy(entity: Entity, user: User, product: Product, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.samples.endpoints.JacksonEndpoint', 'dummy', { entity, user, product }, init);
}

const JacksonEndpoint = {
    dummy,
};

export default JacksonEndpoint;
